### ISSUE TITLE
Regarding the results

### ISSUE THREAD

**Issue by hcrisA** [Quyans/Drag-Your-Gaussian#14]
Hello, I ran the command `python drag_3d.py --config configs/main.yaml --colmap_dir ./data/face/ --gs_source ./data/face/point_cloud.ply --point_dir ./data/face/export_1/drag_points.json --mask_dir ./data/face/export_1/gaussian_mask.pt --output_dir result` However, the result doesn't look very good.

<img width="1552" height="581" alt="Image" src="https://github.com/user-attachments/assets/4c70a4d3-5cdf-4c46-9cec-1ddbf068be86" />

<img width="1538" height="569" alt="Image" src="https://github.com/user-attachments/assets/2603508a-1bc3-49a4-9fc2-3b5071580773" />

I only modified the code  `drag_pipeline.py`  because the `dreamshaper-8-inpainting` in the checkpoints seems to be incompatible with the code.
    
# Handle Inpainting Model (9 channels)
            # Check config from the base unet, handling PeftModel wrapper
            in_channels = 4
            base_model = self.unet_lora
            if hasattr(base_model, "get_base_model"):
                base_model = base_model.get_base_model()
            
            if hasattr(base_model, "config") and hasattr(base_model.config, "in_channels"):
                in_channels = base_model.config.in_channels
            
            if in_channels == 9:
                # Expand input for inpainting model: [latents, mask, masked_latents]
                mask = torch.ones((latents_noisy.shape[0], 1, latents_noisy.shape[2], latents_noisy.shape[3]), device=latents_noisy.device, dtype=latents_noisy.dtype)
                masked_image_latents = torch.zeros((latents_noisy.shape[0], 4, latents_noisy.shape[2], latents_noisy.shape[3]), device=latents_noisy.device, dtype=latents_noisy.dtype)
                latent_model_input = torch.cat([latents_noisy, mask, masked_image_latents], dim=1)
```

# repeat the latent and pad zeros for handle and target embeddings
            if do_classifier_free_guidance:
                latent_model_input = latent_model_input.repeat(2, 1, 1, 1)

            noise_pred = self.unet(
                latent_model_input,
                t,
                encoder_hidden_states=prompt_embeds,
                cross_attention_kwargs=cross_attention_kwargs,
                down_block_additional_residuals=None,
                mid_block_additional_residual=None,
                return_dict=False,
            )[0]
            noise_pred_uncond, noise_pred_cond = noise_pred.chunk(2)
            
            noise_pred = noise_pred_uncond + guidance_scale_points * (
                noise_pred_cond - noise_pred_uncond
            )

            **with torch.cuda.amp.autocast(enabled=True):
                unet_lora_input = torch.cat([latents_noisy, mask, masked_image_latents], dim=1)
                noise_pred_learnable = self.unet_lora(
                    unet_lora_input,
                    t,
                    encoder_hidden_states=self.learnable_embeddings.repeat(batch_size,1,1),
                ).sample**
```

**Comment by hcrisA** [comment]
Now, I have restored the code to its original state and replaced "`dreamshaper-8-inpainting`" with "`botp/stable-diffusion-v1-5-inpainting`", but the result is still not satisfactory.
<img width="1446" height="538" alt="Image" src="https://github.com/user-attachments/assets/d3138ffd-201f-4d0b-a64e-5add41fd577d" />

<img width="1439" height="523" alt="Image" src="https://github.com/user-attachments/assets/ab817246-d088-4b79-8d5f-61f96ea0e98f" />

**Comment by liqing-hub** [comment]
Excuse me, could you please share me your modified code files that fixed the channel mismatch error? I met the same problem and your solution works perfectly. I would really appreciate it!
